pixeltable/io/pandas.py

```diff
@@ -0,0 +1,144 @@
+from typing import Optional, Any, Iterable
+
+import numpy as np
+
+import pandas as pd
+import pixeltable as pxt
+import pixeltable.exceptions as excs
+import pixeltable.type_system as ts
+
+
+def import_pandas(
+        tbl_name: str,
+        df: pd.DataFrame,
+        *,
+        schema: Optional[dict[str, pxt.ColumnType]] = None
+) -> pxt.catalog.InsertableTable:
+    if schema is None:
+        # Infer schema
+        schema = _df_to_pxt_schema(df)
+    else:
+        # Validate the specified schema
+        if len(schema) != len(df.columns):
+            raise excs.Error(
+                f'Specified schema does not match number of columns in the given DataFrame: '
+                f'{len(schema)} != {len(df.columns)}'
+            )
+    tbl_rows = (
+        dict(_df_row_to_pxt_row(row, schema))
+        for row in df.itertuples()
+    )
+    table = pxt.create_table(tbl_name, schema)
+    table.insert(tbl_rows)
+    return table
+
+
+def import_csv(
+        table_path: str,
+        filepath_or_buffer,
+        schema: Optional[dict[str, ts.ColumnType]] = None,
+        **kwargs
+) -> pxt.catalog.InsertableTable:
+    """
+    Creates a new `InsertableTable` from a csv file. This is a convenience method and is equivalent
```

Comment:
Let's reference Table here, not InsertableTable. I don't think the user should have to be aware of the latter.